appshell/appshell_extensions_mac.mm

```diff
@@ -408,23 +409,36 @@ int32 ShowOpenDialog(bool allowMulitpleSelection,
     
     [openPanel setAllowedFileTypes:allowedFileTypes];
     
-    [openPanel beginSheetModalForWindow:[NSApp mainWindow] completionHandler:nil];
-    if ([openPanel runModal] == NSOKButton)
+    // cache the browser and response variables, so that these
+    // can be accessed from within the completionHandler block.
+    CefRefPtr<CefBrowser>        _browser  = browser;
+    CefRefPtr<CefProcessMessage> _response = response;
```

Comment:
I am sure compiler will anyway optimize and will get rid of these two variables anyway.